sirepo/package_data/static/js/cloudmc.js

```diff
@@ -245,23 +248,43 @@ SIREPO.app.controller('GeometryController', function (appState, cloudmcService,
             downloadRemoteGeometryFile();
             return;
         }
+        hasGeometry = true;
         self.simState.runSimulation();
     }
 
+    function verifyGeometry() {
+        requestSender.sendStatefulCompute(
+            appState,
+            (data) => {
+                // don't initialize simulation until geometry is known
+                hasGeometry = data.animationDirExists;
+                self.simState = persistentSimulation.initSimulationState(self);
+            },
+            {
+                method: 'check_animation_dir',
+                args: {
+                    modelName: 'dagmcAnimation',
+                },
+            },
+        );
+    }
+
```

Comment:
I made a few changes:
- don't call initSimulationState() until the run_dir status is known. Otherwise it could make a call to run-status before the stateful compute has finished and think it has volumes to display.
- moved self.isProcessing to a local variable hasGeometry because it isn't referenced outside the controller
- changed stateless compute to stateful compute because it checks state on the server (whether the directory exists)
- Moved the stateful compute argument within an "args" value to keep it out of the top-level argument space.

@gurhar1133 give it a try - I tested in several cases (discard changes, cancel geometry, new simulation)

Reply:
Yes, this is good. Should I merge?